### ISSUE TITLE
the older value didn't chang when user invokes the undo 

### ISSUE THREAD

**Issue by JerryWu1234** [mui/mui-toolpad#2323]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Steps:

1. modify the page
2. first you type a name and then invoke the undo 
3.
![image](https://github.com/mui/mui-toolpad/assets/27261522/a45a53a4-952a-4afb-9947-2583c1666431)

**Comment by JerryWu1234** [comment]
https://github.com/JerryWu1234/mui-toolpad/blob/8d29d72523eaef1b5d8fb466491b4df61e8d4b22/packages/toolpad-app/src/toolpad/AppEditor/NodeNameEditor.tsx#L29

```react
const handleNameCommit = React.useCallback(async () => {
    const oldname = dom.nodes[node.id];
    console.log('undo', oldname.name, nameInput);
    if (isNameValid) {
      if (nameInput !== oldname.name) {
        domApi.setNodeName(node.id, nameInput);
      }
    } else {
      setNameInput(node.name);
    }
    if (isNameValid && oldname.type === 'page' && nameInput !== oldname.name) {
      await client.mutation.deletePage(oldname.name);
    }
  }, [isNameValid, domApi, node.id, node.name, nameInput, dom]);
```

**Comment by Janpot** [comment]
Yes, renaming the page seems to do strange things. This is where our AppDom model conflicts a lot with our file system model. We should stop using the `NodeNameEditor` for the purpose of renaming files and instead have an input that calls a `renamePage` function on the server that renames the actual file on disk. We should not operate on the appdom at all for the case of renaming/creating/deleting files.

**Comment by JerryWu1234** [comment]
> Yes, renaming the page seems to do strange things. This is where our AppDom model conflicts a lot with our file system model. We should stop using the `NodeNameEditor` for the purpose of renaming files and instead have an input that calls a `renamePage` function on the server that renames the actual file on disk. We should not operate on the appdom at all for the case of renaming/creating/deleting files.

@Janpot https://github.com/mui/mui-toolpad/pull/2171
should this pr's renamepage remove to serve side to finish it ?

**Comment by Janpot** [comment]
Yes, I think when the page is renamed on the frontend it should call a function on the backend that renames the folder. And doesn't do anything with the dom at all.
